src/dataclass_binder/_impl.py

```diff
@@ -314,6 +325,15 @@ def _bind_to_single_type(self, value: object, field_type: type, context: str) ->
                 if not isinstance(value, str):
                     raise TypeError(f"Expected TOML string for path '{context}', got '{type(value).__name__}'")
                 return field_type(value)
+            elif issubclass(field_type, ReprEnum):
+                return field_type(value)
```

Comment:
This case is missing the detailed error reporting that the other `Enum` case does have. Also it raises `ValueError` rather than the documented `TypeError`.